mindmeld/components/dialogue.py

```diff
@@ -1011,6 +1027,38 @@ async def call_async(self, request, responder):
         """
         self(request, responder)
 
+    def invoke(self, request, responder):
+        """
+        Invoke slot-filling as a direct call without requiring a decorator.
+        """
+        self._invoked = True
+        kwargs = {'domain': request.domain, 'intent': request.intent}
+
+        # Set dialogue rule to auto_fill and
+        # store current rule for resetting once invoke is complete
+
+        name = self._entrance_handler.__name__
+        _called_from = sys._getframe().f_back.f_code.co_name
```

Comment:
We need to some error checking here. `sys._getframe().f_back` could be `None` for example.

Reply:
I think it will always return between a <function> and a <module> if called from outside. I'll check.